### DISCUSSION TITLE
Self factory reset after DS18B20 sensor connection lost.

### DISCUSSION THREAD

**Discussion by jpier123** [arendst/Tasmota#19926] [Q&A] [answered]
Dear Community,

I'm haveing problem with  ESP32 self-reset after multiple DS18B20 sensor lost. It is happening with or without DS18Alias implemented.

Console log shows multilple sensor raad probelms  just before factory reset:  "MQT: stat/ESPSMD_2F5FA8/RESULT = {"Reset":"Reset and Restarting"}"

ESP goes in to factory state after.

Program Version: 13.2.0(tasmota32)

==================================================
14:01:36.070 Read sensor 0 failed in Ds18x20EverySecond.
14:01:36.085 Read sensor 1 failed in Ds18x20EverySecond.
14:01:36.098 Read sensor 2 failed in Ds18x20EverySecond.
14:01:37.994 Read sensor 1 failed in Ds18x20Show.
14:01:38.007 Read sensor 2 failed in Ds18x20Show.
14:01:38.020 Read sensor 3 failed in Ds18x20Show.
14:01:38.196 Read sensor 0 failed in Ds18x20EverySecond.
14:01:38.209 Read sensor 1 failed in Ds18x20EverySecond.
14:01:38.223 Read sensor 2 failed in Ds18x20EverySecond.
14:01:38.236 Read sensor 3 failed in Ds18x20EverySecond.
14:01:38.551 RUL: COUNTER#C2>500 performs "if(var8!=1) ruletimer5 0;var8 1;Power4 1 endif"
14:01:38.562 MQT: stat/ESPSMD_2F5FA8/RESULT = {"If":"Done"}
14:01:38.834 Read sensor 1 failed in Ds18x20Show.
14:01:38.847 Read sensor 2 failed in Ds18x20Show.
14:01:38.861 Read sensor 3 failed in Ds18x20Show.
14:01:40.069 Read sensor 0 failed in Ds18x20EverySecond.
14:01:40.095 Read sensor 2 failed in Ds18x20EverySecond.
14:01:40.109 Read sensor 3 failed in Ds18x20EverySecond.
14:01:41.256 Read sensor 0 failed in Ds18x20Show.
14:01:41.270 Read sensor 1 failed in Ds18x20Show.
14:01:42.071 Read sensor 1 failed in Ds18x20EverySecond.
14:01:42.085 Read sensor 2 failed in Ds18x20EverySecond.
14:01:42.099 Read sensor 3 failed in Ds18x20EverySecond.
14:01:43.084 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Var6":"0.15"}
14:01:43.752 RUL: COUNTER#C2>500 performs "if(var8!=1) ruletimer5 0;var8 1;Power4 1 endif"
14:01:43.763 MQT: stat/ESPSMD_2F5FA8/RESULT = {"If":"Done"}
14:01:44.026 Read sensor 1 failed in Ds18x20Show.
14:01:46.078 Read sensor 1 failed in Ds18x20EverySecond.
14:01:47.086 Read sensor 0 failed in Ds18x20Show.
14:01:47.100 Read sensor 1 failed in Ds18x20Show.
14:01:47.125 Read sensor 3 failed in Ds18x20Show.
14:01:48.069 Read sensor 0 failed in Ds18x20EverySecond.
14:01:49.345 RUL: COUNTER#C2>500 performs "if(var8!=1) ruletimer5 0;var8 1;Power4 1 endif"
14:01:49.356 MQT: stat/ESPSMD_2F5FA8/RESULT = {"If":"Done"}
14:01:49.625 Read sensor 1 failed in Ds18x20Show.
14:01:50.058 Read sensor 0 failed in Ds18x20Show.
14:01:50.293 Read sensor 0 failed in Ds18x20EverySecond.
14:01:50.307 Read sensor 1 failed in Ds18x20EverySecond.
14:01:50.332 Read sensor 3 failed in Ds18x20EverySecond.
14:01:52.065 Read sensor 0 failed in Ds18x20EverySecond.
14:01:52.091 Read sensor 2 failed in Ds18x20EverySecond.
14:01:52.105 Read sensor 3 failed in Ds18x20EverySecond.
14:01:53.076 Read sensor 1 failed in Ds18x20Show.
14:01:53.090 Read sensor 2 failed in Ds18x20Show.
14:01:53.139 Read sensor 6 failed in Ds18x20Show.
14:01:54.077 Read sensor 1 failed in Ds18x20EverySecond.
14:01:54.103 Read sensor 3 failed in Ds18x20EverySecond.
14:01:54.140 Read sensor 6 failed in Ds18x20EverySecond.
14:01:54.745 RUL: COUNTER#C2>500 performs "if(var8!=1) ruletimer5 0;var8 1;Power4 1 endif"
14:01:54.756 MQT: stat/ESPSMD_2F5FA8/RESULT = {"If":"Done"}
14:01:55.029 Read sensor 1 failed in Ds18x20Show.
14:01:55.054 Read sensor 3 failed in Ds18x20Show.
14:01:55.092 Read sensor 6 failed in Ds18x20Show.
14:01:56.062 MQT: tele/ESPSMD_2F5FA8/STATE = {"Time":"2023-11-04T14:01:56","Uptime":"0T00:10:25","UptimeSec":625,"Heap":93,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":1,"Berry":{"HeapUsed":4,"Objects":43},"POWER1":"OFF","POWER2":"ON","POWER3":"OFF","POWER4":"ON","POWER5":"ON","POWER6":"ON","POWER7":"OFF","POWER8":"OFF","POWER9":"OFF","POWER10":"OFF","POWER11":"OFF","POWER12":"OFF","POWER13":"OFF","POWER14":"OFF","POWER15":"OFF","POWER16":"OFF","POWER17":"OFF","POWER18":"OFF","POWER19":"ON","Dimmer":90,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"SS01","BSSId":"E8:65:D4:47:EF:80","Channel":7,"Mode":"11n","RSSI":100,"Signal":-5,"LinkCount":1,"Downtime":"0T00:00:12"}}
14:01:56.147 RUL: TELE-WIFI#RSSI performs "publish domoticz/in {"idx":144,"svalue":"100"}"
14:01:56.156 MQT: domoticz/in = {"idx":144,"svalue":"100"}
14:01:56.184 Read sensor 0 failed in Ds18x20Show.
14:01:56.209 Read sensor 2 failed in Ds18x20Show.
14:01:56.222 Read sensor 3 failed in Ds18x20Show.
14:01:56.260 Read sensor 6 failed in Ds18x20Show.
14:01:56.333 MQT: tele/ESPSMD_2F5FA8/SENSOR = {"Time":"2023-11-04T14:01:56","COUNTER":{"C1":4432,"C2":5912},"DS18B20-2":{"Id":"3CE104579322","Temperature":8.9},"DS18B20-5":{"Id":"3CE104571458","Temperature":22.8},"DS18B20-6":{"Id":"3CE10457B9F8","Temperature":24.3},"DS18B20-8":{"Id":"3CE10457E3AF","Temperature":26.4},"DS18B20-9":{"Id":"3C51F649B890","Temperature":22.4},"DS18B20-10":{"Id":"3CA0F649311A","Temperature":18.9},"DS18B20-11":{"Id":"3CE10457B8CF","Temperature":19.3},"DS18B20-12":{"Id":"E54B620E64FF","Temperature":35.9},"Thermostat0":{"ThermostatModeSet":0,"TempTargetSet":22.50,"CtrDutyCycleRead":0,"ControlMethod":0,"HybridControllerPhase":1,"EmergencyState":0},"TempUnit":"C"}
14:01:57.311 MQT: tele/ESPSMD_2F5FA8/BLE = {"Time":"2023-11-04T14:01:57","BLEDevices":{"total":0}}
14:01:57.359 MQT: tele/ESPSMD_2F5FA8/BLE = {"Time":"2023-11-04T14:01:57","BLE":{"scans":0,"adverts":0,"devices":0,"resets":0}}
14:01:57.417 Read sensor 0 failed in Ds18x20EverySecond.
14:01:57.431 Read sensor 1 failed in Ds18x20EverySecond.
14:01:57.444 Read sensor 2 failed in Ds18x20EverySecond.
14:01:57.458 Read sensor 3 failed in Ds18x20EverySecond.
14:01:57.495 Read sensor 6 failed in Ds18x20EverySecond.
14:01:57.657 Read sensor 6 failed in Ds18x20Show.
14:01:57.786 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Var2":"22.8"}
14:01:58.005 MQT: stat/ESPSMD_2F5FA8/RESULT = {"DisplayText":"[s1f1l1c7]22.9"}
14:01:58.065 Read sensor 0 failed in Ds18x20EverySecond.
14:01:58.079 Read sensor 1 failed in Ds18x20EverySecond.
14:01:58.093 Read sensor 2 failed in Ds18x20EverySecond.
14:01:58.106 Read sensor 3 failed in Ds18x20EverySecond.
14:01:58.143 Read sensor 6 failed in Ds18x20EverySecond.
14:01:58.221 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Var6":"0.21"}
14:01:58.450 MQT: stat/ESPSMD_2F5FA8/RESULT = {"DisplayText":"[s1f1l1c13]Q0.00m3"}
14:01:58.652 MQT: stat/ESPSMD_2F5FA8/RESULT = {"DisplayText":"[s1f1l2c13]P0.00kW"}
14:01:58.855 MQT: stat/ESPSMD_2F5FA8/RESULT = {"DisplayText":"[s1f1l3c13]C0.00"}
14:01:59.140 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Var3":"26.4"}
14:01:59.356 MQT: stat/ESPSMD_2F5FA8/RESULT = {"DisplayText":"[s1f1l2c1]23.7"}
14:01:59.547 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Var4":"24.3"}
14:01:59.761 MQT: stat/ESPSMD_2F5FA8/RESULT = {"DisplayText":"[s1f1l2c7]24.4"}
14:01:59.958 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Mem1":"0.80"}
14:02:00.096 Read sensor 0 failed in Ds18x20EverySecond.
14:02:00.111 Read sensor 1 failed in Ds18x20EverySecond.
14:02:00.136 Read sensor 3 failed in Ds18x20EverySecond.
14:02:00.173 Read sensor 6 failed in Ds18x20EverySecond.
14:02:00.247 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Mem2":"-1.50"}
14:02:00.451 MQT: domoticz/in = {"idx":100,"svalue":"0.90"}
14:02:00.658 MQT: domoticz/in = {"idx":101,"svalue":"-1.50"}
14:02:00.863 MQT: domoticz/in = {"idx":133,"svalue":"0.00"}
14:02:01.041 MQT: domoticz/in = {"idx":298,"svalue":"10.00"}
14:02:01.089 Read sensor 0 failed in Ds18x20Show.
14:02:01.103 Read sensor 1 failed in Ds18x20Show.
14:02:01.117 Read sensor 2 failed in Ds18x20Show.
14:02:01.130 Read sensor 3 failed in Ds18x20Show.
14:02:01.167 Read sensor 6 failed in Ds18x20Show.
14:02:01.290 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Mem3":"985.33"}
14:02:01.496 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Mem4":"0.23"}
Read sensor 2 failed in Ds18x20EverySecond.
14:02:10.107 Read sensor 3 failed in Ds18x20EverySecond.
14:02:10.144 Read sensor 6 failed in Ds18x20EverySecond.
14:02:11.039 Read sensor 0 failed in Ds18x20Show.
14:02:11.053 Read sensor 1 failed in Ds18x20Show.
14:02:11.066 Read sensor 2 failed in Ds18x20Show.
14:02:11.116 Read sensor 6 failed in Ds18x20Show.
14:02:12.064 Read sensor 0 failed in Ds18x20EverySecond.
14:02:12.078 Read sensor 1 failed in Ds18x20EverySecond.
14:02:12.104 Read sensor 3 failed in Ds18x20EverySecond.
14:02:12.140 Read sensor 6 failed in Ds18x20EverySecond.
14:02:12.983 RUL: COUNTER#C2>500 performs "if(var8!=1) ruletimer5 0;var8 1;Power4 1 endif"
14:02:12.994 MQT: stat/ESPSMD_2F5FA8/RESULT = {"If":"Done"}
14:02:13.056 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Var6":"0.63"}
14:02:13.357 Read sensor 0 failed in Ds18x20Show.
14:02:13.370 Read sensor 1 failed in Ds18x20Show.
14:02:13.395 Read sensor 3 failed in Ds18x20Show.
14:02:13.432 Read sensor 6 failed in Ds18x20Show.
14:02:14.019 Read sensor 0 failed in Ds18x20Show.
14:02:14.033 Read sensor 1 failed in Ds18x20Show.
14:02:14.047 Read sensor 2 failed in Ds18x20Show.
14:02:14.060 Read sensor 3 failed in Ds18x20Show.
14:02:14.097 Read sensor 6 failed in Ds18x20Show.
14:02:14.227 Read sensor 0 failed in Ds18x20EverySecond.
14:02:14.241 Read sensor 1 failed in Ds18x20EverySecond.
14:02:14.266 Read sensor 3 failed in Ds18x20EverySecond.
14:02:14.304 Read sensor 6 failed in Ds18x20EverySecond.
14:02:16.060 Read sensor 0 failed in Ds18x20EverySecond.
14:02:16.074 Read sensor 1 failed in Ds18x20EverySecond.
14:02:16.088 Read sensor 2 failed in Ds18x20EverySecond.
14:02:16.102 Read sensor 3 failed in Ds18x20EverySecond.
14:02:16.138 Read sensor 6 failed in Ds18x20EverySecond.
14:02:16.282 MQT: stat/ESPSMD_2F5FA8/RESULT = **{"Reset":"Reset and Restarting"}**

============================================================

**Comment by jpier123** [comment]
15:12:07.586 MQT: stat/ESPSMD_2F5FA8/POWER4 = ON
15:12:08.352 Read sensor 0 failed in Ds18x20EverySecond.
15:12:08.377 Read sensor 2 failed in Ds18x20EverySecond.
15:12:08.390 Read sensor 3 failed in Ds18x20EverySecond.
15:12:09.744 Read sensor 2 failed in Ds18x20Show.
15:12:09.758 Read sensor 3 failed in Ds18x20Show.
15:12:10.384 Read sensor 2 failed in Ds18x20EverySecond.
15:12:10.398 Read sensor 3 failed in Ds18x20EverySecond.
15:12:11.883 RUL: COUNTER#C2>500 performs "if(var8!=1) ruletimer5 0;var8 1;Power4 1 endif"
15:12:11.894 MQT: stat/ESPSMD_2F5FA8/RESULT = {"If":"Done"}
15:12:12.160 Read sensor 1 failed in Ds18x20Show.
15:12:12.360 Read sensor 0 failed in Ds18x20EverySecond.
15:12:12.374 Read sensor 1 failed in Ds18x20EverySecond.
15:12:12.387 Read sensor 2 failed in Ds18x20EverySecond.
15:12:12.400 Read sensor 3 failed in Ds18x20EverySecond.
15:12:13.122 Read sensor 1 failed in Ds18x20Show.
15:12:13.136 Read sensor 2 failed in Ds18x20Show.
15:12:13.150 Read sensor 3 failed in Ds18x20Show.
15:12:14.372 Read sensor 1 failed in Ds18x20EverySecond.
15:12:14.386 Read sensor 2 failed in Ds18x20EverySecond.
15:12:14.400 Read sensor 3 failed in Ds18x20EverySecond.
15:12:15.645 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Var6":"0.58"}
15:12:16.132 Read sensor 0 failed in Ds18x20Show.
15:12:16.146 Read sensor 1 failed in Ds18x20Show.
15:12:16.160 Read sensor 2 failed in Ds18x20Show.
15:12:16.369 Read sensor 1 failed in Ds18x20EverySecond.
15:12:16.590 MQT: stat/ESPSMD_2F5FA8/RESULT = {"Reset":"Reset and Restarting"}
15:12:16.774 CFG: Use defaults


How to stop  {"Reset":"Reset and Restarting"}?

**Reply by sfromis** [reply, marked as answer]
You have to fix the badly configured button pin. A button registering as "pushed" will by default trigger a recovery after 40 seconds.

**Reply by jpier123** [reply]
Can self reset be blocked or de-activated?

**Reply by sfromis** [reply]
It could, but just fix the bad config. Use a Switch pin instead of Button if it is expected to be engaged for a long time. Maybe it would make sense to have it inverted, depending on what you have on the pin.

**Reply by jpier123** [reply]
Again 100% right . Used was bad pcf8574.dat template file. 

Thanks one more time.

Best Regards, 
JP
